bool proceed
---
bool nonext
int32 next
int32 priority